pretrain_gpt.py

```diff
@@ -28,14 +28,23 @@
 from torch import nn
 import torch.nn.functional as F
 
-def model_provider(pre_process=True, post_process=True):
+
+def model_provider(pre_process=True, post_process=True, ckpt_transfer_model=False):
     """Build the model."""
 
     print_rank_0('building GPT model ...')
     see_memory_usage(f"Before Building Model", force=True)
 
     args = get_args()
     config = core_transformer_config_from_args(args)
+    
+    if ckpt_transfer_model:
+        return GPTModel(config=config,
+                    num_tokentypes=0,
+                    parallel_output=True,
+                    pre_process=pre_process,
+                    post_process=post_process)
+    
     with deepspeed.zero.Init(sequence_data_parallel_group=mpu.get_sequence_data_parallel_group(),
```

Comment:
there must be some better solution to init model without init distibute group. please help me ..

Reply:
The distributed initialization only occurs for `args.zero_stage==3`. Have you tried with different stage value on command line?

Reply:
> The distributed initialization only occurs for `args.zero_stage==3`. Have you tried with different stage value on command line?

The problem is `mpu.get_sequence_data_parallel_group()`. How can I solve this problem?
```
  File "/mnt/megatron-deepspeed/pretrain_gpt.py", line 48, in model_provider
    with deepspeed.zero.Init(sequence_data_parallel_group=mpu.get_sequence_data_parallel_group(),
  File "/mnt/megatron-deepspeed/megatron/core/parallel_state.py", line 369, in get_sequence_data_parallel_group
    assert _SEQUENCE_DATA_PARALLEL_GROUP is not None, \
AssertionError: sequence data parallel group is not initialized
```